lib/utilities.js

```diff
@@ -293,7 +294,51 @@ module.exports.dispositionRange = function(message) {
  * @param [policyOverrides]   link creation policy overrides
  * @return {String}
  */
-module.exports.linkName = function(address, policyOverrides) {
+utilities.linkName = function(address, policyOverrides) {
   return ((!!policyOverrides && !!policyOverrides.name) ?
     policyOverrides.name : address + '_' + uuid.v4());
 };
+
+
+/**
+ * Assign properties given to dest IFF they are defined in source
+ */
+utilities.assignDefined = function(dest, source, properties) {
+  var propertiesToAssign = _.omit(properties, function(value, key) {
+    return _.isNull(source[key]) || _.isUndefined(source[key]);
+  });
+
+  _.assign(dest, propertiesToAssign);
+};
+
+/**
+ * Extract values for a DescribedType based on that types EncodeOrdering
+ */
+utilities.extractDescribedType = function(Type, value, defaults) {
```

Comment:
This method doesn't pull its own weight - if you were to pass `{}` in as the _"instance"_ to `assignDescribedTypeFromPayload` it would do the exact same thing.